editor/components/block-list/block.js

```diff
@@ -187,6 +193,14 @@ export class BlockListBlock extends Component {
 		document.removeEventListener( 'mousemove', this.stopTypingOnMouseMove );
 	}
 
+	updateHasEditableContent() {
+		// Update the hasEditableContent property
+		const hasEditableContent = some( focus.tabbable.find( this.node ), isEditableNode );
```

Comment:
Could we... not query select and filter on every change to the block? If it's a matter of avoiding effects that only appear on focusing a specific element (`.editor-block-list__block-edit`), could we do it on that element's focus event?

Reply:
It could work, but the behavior of the block could change if its content change as well Imagine an image transformed to a text somehow or I don't know, blocks can do what they want.

Reply:
In the absolute worst case I'd still prefer a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) monitoring `childList` and `attributes`. I expect at least this wouldn't fire the callback on every keypress.

Reply:
Updated using a MutationObserver. Let me know what you think. In my testing, it doesn't trigger on each keypress.